app/main.py

```diff
@@ -1,13 +1,42 @@
 def format_linter_error(error: dict) -> dict:
-    # write your code here
-    pass
-
+    return {"line": error["line_number"],
+    "column": error["column_number"],
+    "message": error["text"],
+    "name": error["code"],
+    "sourse": "flake8"}
 
 def format_single_linter_file(file_path: str, errors: list) -> dict:
-    # write your code here
-    pass
+    return {"errors":
+    [
+        {
+            "line": error["line_number"],
+            "column": error["column_number"],
+            "message": error["text"],
+            "name": error["code"],
+            "sourse": "flake8"
+            }
+            for error in errors],
+            "path": file_path,
+            "status": "failed"
+            if len(errors) != 0
+            else {"errors": [], "status": "passed", "path": file_path}}
```

Comment:
The logic for determining the 'status' is incorrect. The 'else' clause is attempting to return a dictionary, but the 'if' clause is already returning a dictionary. Consider restructuring this logic to ensure consistent return types.